### PR TITLE
[2135] Allow TreeItemContextMenuContributions to only apply to specific trees

### LINKED ISSUE #2135: TreeItemContextMenuContribution should be Tree-specific
It's possible to contribute context menu entries to be added to TreeItems, but this is done in a global context (`TreeItemContextMenuContext`) and `TreeItemContextMenuContributionProps#canHandle` is only given the `TreeItem`.
From that it is not possible to discriminate between different trees (not possible at the moment, but coming soon with #2126).

Tree item actions which make sense in the global Explorer may not make sense in other Tree representations (and vice versa).

### PR DESCRIPTION
**Pull request by pcdavid**
Bug: https://github.com/eclipse-sirius/sirius-components/issues/2135
Signed-off-by: Pierre-Charles David <pierre-charles.david@obeo.fr>

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -755,8 +755,8 @@ When closed, clicking on any of the views' icon will re-open the panel to make t
 === Bug fixes
 
 - https://github.com/eclipse-sirius/sirius-components/issues/992[#992] [view] Let the `ViewValidator` consider statically contributed `EPackages` when validating domain types
-- https://github.com/eclipse-sirius/sirius-components/issues/991[#991] [diagram] Restore edge creation tools feedback
-- https://github.com/eclipse-sirius/sirius-components/issues/1056[#1056] [core] Fix an invalid usage of `forwardRef` in  `DiagramTreeItemContextMenuContribution`
+- https://github.com/eclipse-sirius/sirius-components/issues/991[#991] [diagram] Restore edge creation tools feedbackDiagramTreeItemContextMenuContribution
+- https://github.com/eclipse-sirius/sirius-components/issues/1056[#1056] [core] Fix an invalid usage of `forwardRef` in  ``
```
Comment (frouene) [inline CHANGELOG.adoc]: Mishandling ?

Comment (pcdavid) [reply CHANGELOG.adoc]: Fixed.

**Review by AxelRICHARD — APPROVED** [review]

File: packages/sirius-web/frontend/sirius-web/src/views/edit-project/EditProjectView.tsx [hunk 2]
```diff
@@ -134,15 +134,21 @@ export const EditProjectView = () => {
 
     const treeItemContextMenuContributions = [
       <TreeItemContextMenuContribution
-        canHandle={(item: GQLTreeItem) => item.kind.startsWith('siriusWeb://document')}
+        canHandle={(treeId: string, item: GQLTreeItem) =>
+          treeId?.startsWith('explorer://') && item.kind.startsWith('siriusWeb://document')
```
Comment (sbegaudeau) [inline packages/sirius-web/frontend/sirius-web/src/views/edit-project/EditProjectView.tsx]: Why use `?.`, the type is not optional, same comment for the changes below

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

File: packages/sirius-web/frontend/sirius-web/src/views/edit-project/EditProjectView.tsx (hunk 2 above)
Comment (pcdavid) [reply packages/sirius-web/frontend/sirius-web/src/views/edit-project/EditProjectView.tsx]: Done.

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**